### PR TITLE
gh-107862: Add roundtrip `hypothesis` tests to `test_binascii`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-107862

### CODE REVIEW

**Comment by ambv** [comment]
Closing and re-opening to retrigger CLA checks. Sorry for the noise.

File: Lib/test/test_binascii.py [hunk 1]
```diff
@@ -259,6 +275,12 @@ def test_hex(self):
         self.assertEqual(binascii.hexlify(self.type2test(s)), t)
         self.assertEqual(binascii.unhexlify(self.type2test(t)), u)
 
+    @hypothesis.given(binary=hypothesis.strategies.binary())
+    def test_hex_roundtrip(self, binary):
+        converted = binascii.hexlify(self.type2test(binary))
+        restored = binascii.unhexlify(self.type2test(converted))
```
Comment (hauntsaninja) [inline Lib/test/test_binascii.py:281]: Wonder if unhexlify should also take a sep and bytes_per_sep argument

**Review by hauntsaninja — APPROVED** [review]
Thanks for working on this!

**Comment by sobolevn** [comment]
Btw, @rhettinger was against doing this.

**Comment by hauntsaninja** [comment]
I agree that we shouldn't rely on Hypothesis' strategies as a substitute for fixed tests.

But I think the implementation chosen when adding property based tests in CPython is quite sensible! None of the Hypothesis strategies run by default. It's used as a DSL to write property based tests where `@hypothesis.example` is used to provide the fixed test cases. However, you can then opt-in to get Hypothesis to run its strategies against your existing test.

For this specific PR, I guess maybe I should have asked Nikita to add some specific `@hypothesis.example` here. Raymond, I'm happy to open a PR adding `example`s, or to revert, or to let it be. Just let me know!

(I also obviously agree we shouldn't use Hypothesis to write tests to enforce properties that are not actually properties, like the colorsys example)

**Pull request merged**